src/core/build.jl

```diff
@@ -269,7 +270,27 @@ function _build(comp_def::AbstractCompositeComponentDef,
     # @info "  par_dict $(par_dict)"
 
     comps = [_build(cd, var_dict, par_dict, time_bounds) for cd in compdefs(comp_def)]
-    return CompositeComponentInstance(comps, comp_def, time_bounds)
+
+    variables = _get_variables(comp_def, var_dict)
+    parameters = _get_parameters(comp_def, par_dict)
+
+    return CompositeComponentInstance(comps, comp_def, time_bounds, variables, parameters)
+end
+
+# helper functions for to create the variables and parameters NamedTuples for a 
+# CompositeComponentInstance
+function _get_variables(comp_def::AbstractCompositeComponentDef, var_dict::Dict{ComponentPath, Any})
+
+    # TODO (dummy below)
+    NT = NamedTuple(Tuple(:var1, :var2), Tuple(1.0, 2.0))
```

Comment:
@ckingdon95 advice here would be great!

Reply:
so all this needs to do is loop through the comp_def's namespace (which is a dictionary) and keep everything that is a CompositeVariableDef. Then the symbols in the NameTuple constructor will just be the keys from the namespace, and the values in the second half of the constructor will be the vardef.ref (which gives an UnnamedReference object) from each vardef

```
# pseudo code

var_defs = filter(isa(CompositeVariableDef), com_pdef.namespace)  # this probably doesn't quite work for filtering a dictionary on it's values
symbols = [k for (k, v) in var_defs]
vals = [v.ref for (k, v) in var_defs]
NT = NamedTuple(symbols, vals)